tasks/nunjucks.js

```diff
@@ -35,7 +35,7 @@ module.exports = function(grunt) {
 
         async.each(this.files, function(f, done) {
             var filepath = path.join(process.cwd(), f.src[0]);
-            var data = options.data || {};
+            var data = JSON.parse(JSON.stringify(options.data || {}));
```

Comment:
hm, I'm also not sure, but maybe it will work better and be more clear code?

``` javascript
var data = Object.create(options.data || {});
```

Didn't test it, to be honest, but it should clone data too.